src/main/java/org/mskcc/cmo/messaging/impl/JSGatewayImpl.java

```diff
@@ -297,4 +296,43 @@ public PublishingQueueTask(String msgId, String subject, Object payload) {
             this.payload = payload;
         }
     }
+
+    @Override
+    public String request(String subject, Object message) {
+        if (natsConnection == null) {
+            LOG.error("Error initializing NATS connection");
+        }
+        try {
+            String msg = mapper.writeValueAsString(message);
+            Message reply = natsConnection.request(subject, msg.getBytes(), Duration.ofSeconds(10));
```

Comment:
Is 10 seconds sufficient amount of time to wait for a reply?

Reply:
duration should probably be a property